diff --git a/src/linux/btop_collect.cpp b/src/linux/btop_collect.cpp
index 64eaffe..f3687c2 100644
--- a/src/linux/btop_collect.cpp
+++ b/src/linux/btop_collect.cpp
@@ -224,6 +224,7 @@ namespace Gpu {
 		bool shutdown();
 		template <bool is_init> bool collect(gpu_info* gpus_slice);
 		vector<nvmlDevice_t> devices;
+		vector<fs::path> tegra_load_paths;
 		unsigned int device_count = 0;
 	}
 
@@ -1295,6 +1296,7 @@ namespace Gpu {
 
 			if (device_count > 0) {
 				devices.resize(device_count);
+				tegra_load_paths.resize(device_count);
 				gpus.resize(device_count);
 				gpu_names.resize(device_count);
 
@@ -1347,6 +1349,11 @@ namespace Gpu {
 							gpu_names[i] = s_replace(gpu_names[i], brand, "");
 						}
 						gpu_names[i] = trim(gpu_names[i]);
+						if (gpu_names[i].find("(nvgpu)") != string::npos) {
+							const fs::path tegra_load_path = "/sys/devices/platform/17000000.gpu/load";
+							if (fs::exists(tegra_load_path) and access(tegra_load_path.c_str(), R_OK) == 0)
+								tegra_load_paths[i] = tegra_load_path;
+						}
         			}
 
     				//? Power usage
@@ -1393,16 +1400,22 @@ namespace Gpu {
 				// DebugTimer nvTimer("Nv utilization");
 				//? GPU & memory utilization
 				if (gpus_slice[i].supported_functions.gpu_utilization) {
-					nvmlUtilization_t utilization;
-					result = nvmlDeviceGetUtilizationRates(devices[i], &utilization);
-    				if (result != NVML_SUCCESS) {
-						Logger::warning("NVML: Failed to get GPU utilization: {}", nvmlErrorString(result));
-						if constexpr(is_init) gpus_slice[i].supported_functions.gpu_utilization = false;
+					if (not tegra_load_paths[i].empty()) {
+						const long long load = stoll(readfile(tegra_load_paths[i], "0"));
+						gpus_slice[i].gpu_percent.at("gpu-totals").push_back(clamp((load + 5) / 10, 0ll, 100ll));
 						if constexpr(is_init) gpus_slice[i].supported_functions.mem_utilization = false;
-    				} else {
-						gpus_slice[i].gpu_percent.at("gpu-totals").push_back((long long)utilization.gpu);
-						gpus_slice[i].mem_utilization_percent.push_back((long long)utilization.memory);
-    				}
+					} else {
+						nvmlUtilization_t utilization;
+						result = nvmlDeviceGetUtilizationRates(devices[i], &utilization);
+						if (result != NVML_SUCCESS) {
+							Logger::warning("NVML: Failed to get GPU utilization: {}", nvmlErrorString(result));
+							if constexpr(is_init) gpus_slice[i].supported_functions.gpu_utilization = false;
+							if constexpr(is_init) gpus_slice[i].supported_functions.mem_utilization = false;
+						} else {
+							gpus_slice[i].gpu_percent.at("gpu-totals").push_back((long long)utilization.gpu);
+							gpus_slice[i].mem_utilization_percent.push_back((long long)utilization.memory);
+						}
+					}
 				}
 
 				// nvTimer.stop_rename_reset("Nv clock");
